File: 1272-remove-interval/1272-remove-interval.py
class Solution:
    def removeInterval(self, intervals: List[List[int]], toBeRemoved: List[int]) -> List[List[int]]:
        answer = []
        for i in intervals:
            if i[1] <= toBeRemoved[0]:
                answer.append(i)
            elif i[0] < toBeRemoved[0]:
                answer.append([i[0],toBeRemoved[0]])
            if i[0] >= toBeRemoved[1]:
                answer.append(i)
            elif i[1] > toBeRemoved[1]:
                answer.append([toBeRemoved[1],i[1]])
        return answer

File: 1272-remove-interval/1272-remove-interval.java
class Solution {
    public List<List<Integer>> removeInterval(int[][] intervals, int[] toBeRemoved) {
        List<List<Integer>> resIntervals = new LinkedList<>();
        
        for (int[] interval : intervals) {
            //left non-overlapping
            if (interval[1] <= toBeRemoved[0]) 
                resIntervals.add(Arrays.asList(interval[0], interval[1]));
            else if (interval[0] >= toBeRemoved[1]) 
                resIntervals.add(Arrays.asList(interval[0], interval[1]));
            else{
                if (interval[0] < toBeRemoved[0] && interval[1] > toBeRemoved[1]) {
                    resIntervals.add(Arrays.asList(interval[0], toBeRemoved[0]));
                    resIntervals.add(Arrays.asList(toBeRemoved[1], interval[1]));
                } else if (interval[0] < toBeRemoved[0]) {
                    interval[1] = toBeRemoved[0];
                    resIntervals.add(Arrays.asList(interval[0], interval[1]));
                } else if (interval[1] > toBeRemoved[1]) {
                    interval[0] = toBeRemoved[1];
                    resIntervals.add(Arrays.asList(interval[0], interval[1]));
                }
            }
        }
        
        return resIntervals;
        
        
    }
}